Query Browser: Add Developer perspective version

kyoto opened the pull request:
When a project is selected, send PromQL queries through the Prometheus
tenancy proxy to limit results to the project's namespace.

When in the Developer perspective
  - Show Project dropdown at top of page (like other Developer
    perspective pages)
  - Show "Tech Preview" badge
  - Don't show link to Prometheus UI

### Developer perspective
![screenshot-1](https://user-images.githubusercontent.com/460802/66455165-f0fa9180-eaa4-11e9-897f-a5d4d232e2ed.png)
![screenshot](https://user-images.githubusercontent.com/460802/66454930-44201480-eaa4-11e9-9ee2-707308d4f070.png)

### Administrator perspective (unchanged by this PR)
![screenshot](https://user-images.githubusercontent.com/460802/65690009-1781fb00-e0a9-11e9-83fa-d8df03377b48.png)

spadgett commented:
/hold cancel

kyoto commented:
@spadgett, @christianvogt, @cshinn This PR is now ready for review.

christianvogt commented:
@sspeiche  @serenamarie125 please have a look.
@kyoto why is the tech preview badge applied in dev console but I don't see it in the admin?

kyoto commented:
@christianvogt The Administrator perspective version is already part of 4.2 and not in tech preview. The Developer perspective version relies on the new service monitoring extension for OpenShift, which is in tech preview.

christianvogt commented:
@kyoto the tech preview badges in designs are always on the right:
https://docs.google.com/document/d/1kBt_OeYtTINJKN0xThCJZbChZjSsqEJ-viX-KIaDTdI/edit#heading=h.htexe3yn5kwl

@serenamarie125 this case isn't covered where there are no breadcrumbs  reserving space. Should it still be above the actions dropdown?

serenamarie125 commented:
@kyoto I have a couple comments/questions regarding the experience in the dev perspective

1. @sspeiche I think we discussed this being under Advanced for now right?
1. does the list of items displayed for auto-completion change ?  I'm assuming that users with non cluster-admin access will only have access to a subset of the metrics?
1. this page should always be in context of a project, the secondary masthead needs to be present, and if All Projects is selected we should show the project list to be sure that the user selects a project 

Do we want to meet to discuss these changes?

sspeiche commented:
Where it lands in the navigation, we can change it later.
Putting under Advanced sounds right to me for now.

I'm thinking we could have a top-level item called `Monitoring` or something and then have 2 items `Dashboard` and `Query Browser`

kyoto commented:
/retest

kyoto commented:
@serenamarie125 Thanks for the comments.

> 1. I think we discussed this being under Advanced for now right?

Now moved under Advanced

> 2. does the list of items displayed for auto-completion change ?  I'm assuming that users with non cluster-admin access will only have access to a subset of the metrics?

Thanks for bringing this up. The Monitoring team needs to have some more discussion about this. For now, I have removed the autocomplete options and the metrics list dropdown from the Developer perspective version, so we can move ahead with this PR and address the autocompletions in a separate PR.

> 3. in the case of your second mockup, the secondary masthead should be present.

I've now added the Project selection bar to that page too. Thanks.

> 4. since this page is always in context of a project, does the user need to add the project/namespace name into the query, or do you guys do that automatically?

The user does not need to add the namespace to the query - it is inserted automatically on the backend.

kyoto commented:
Changed to keep autocompletion enabled, but without metrics (just for aggregation operators and functions).

![screenshot](https://user-images.githubusercontent.com/460802/66532801-b0a21e80-eb4b-11e9-9976-0472a8cbf0ee.png)

kyoto commented:
@christianvogt @spadgett I believe I have addressed all comments, could one of you review the code?

christianvogt commented:
> @christianvogt I left the "Tech Preview" badge where it was for now because I understand its position is being discussed by the UX team. I will address with a follow up PR based on the outcome of that discussion.

@kyoto I'm fine with adjusting the location later. But where are these discussions being captured?

christianvogt commented:
/lgtm

christianvogt commented on frontend/packages/dev-console/src/components/MetricsPage.tsx:
```diff
@@ -0,0 +1,35 @@
+  );
+};
+
+export default withStartGuide(MetricsPage);
```
@kyoto I just noticed that you need to wrap the inner component with `withStartGuide` otherwise the namespace bar will be _below_ the start guide header.

christianvogt commented:
/lgtm cancel

christianvogt commented:
Turns out we got other issues around start guide use in dev console. We filled a bug to tackle then. We can get this in and then do a consist update across dev console for all areas.

christianvogt commented:
/lgtm

cshinn commented:
> > @christianvogt I left the "Tech Preview" badge where it was for now because I understand its position is being discussed by the UX team. I will address with a follow up PR based on the outcome of that discussion.
> 
> @kyoto I'm fine with adjusting the location later. But where are these discussions being captured?

@christianvogt I brought this up at the design meeting a few days ago and it seems like there's work being done by one of our visual designers to solve a number of the issues with this design (looks like a button, placement, etc.)I think any effort to implement that will have to include every place that it appears in the app and so I'd consider it outside the scope of this PR. Personally, I think that having the badge next to the page title in this case does a good job of illustrating the scope of what is tech preview and it makes it look less like an action button.

kyoto commented:
/hold

kyoto commented:
@christianvogt Sorry, I needed to rebase. Could you LGTM again?

kyoto commented:
/retest

christianvogt commented:
/lgtm

kyoto commented:
/retest

kyoto commented:
/hold cancel
